src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -2007,6 +1942,48 @@ void CheckClosed()
         }
 
         #endregion
+
+        #region SafeColumnRead
+
+        TValue SafeColumnRead<TValue>(Func<TValue> func)
+            => SafeColumnRead(() => new ValueTask<TValue>(func()), false).GetAwaiter().GetResult();
+
+        ValueTask<TValue> SafeColumnRead<TValue>(Func<ValueTask<TValue>> func)
+            => SafeColumnRead(func, true);
+
+        async ValueTask<TValue> SafeColumnRead<TValue>(Func<ValueTask<TValue>> func, bool async)
+        {
+            var position = Buffer.ReadPosition;
+            try
+            {
+                return await func();
+            }
+            catch (NpgsqlSafeReadException e)
+            {
+                await SkipRemaining();
+                throw e.OriginalException;
```

Comment:
There is a difference between `throw` and `throw catchedException`. The first one do not alter the existing stack trace of an exception, but the second one completely overwrites it making it's impossible to detect the root without debugging. In cases when you need to throw an inner exception use `ExceptionDispatchInfo.Throw`.

Reply:
Good remark! I have learnt something new today. Thanks 😃